pyiron_atomistics/atomistics/structure/neighbors.py

```diff
@@ -1081,3 +1083,101 @@ def get_volume_of_n_sphere_in_p_norm(n=3, p=2):
     """
     return (2*gamma(1+1/p))**n/gamma(1+n/p)
 
+
+class NeighborsTraj(DataContainer):
+
+    """
+    This class generates the neighbors for a given atomistic trajectory. The assumption here is that the trajectory is
+    canonical (no change in the number and types of species throughout the trajectory). The resulting indices,
+    distances, and vectors are stored as numpy arrays.
+
+    """
+
```

Comment:
```suggestion
def __new__(cls, *args, **kwargs):
    instance = super().__new__(cls, *args, **kwargs)
    object.__setattr__(instance, "_has_structure", None)
    return instance
```
This will make sure the `HasStructure` instances is not saved in the container and will therefore not be written to HDF twice.  On the other hand this creates the problem, that when loading a `NeighborsTraj` it can't recreate it. 

One solution for atomistic jobs and the structure container would be to save the job id in the container instead of the whole instance, but that wouldn't work for `Trajectories` since they don't have an ID.

Another thing I can think of is to have `_get_neighbors` become a method of `NeighborsTraj` and have it populate positions, indices and cells attributes as it works on the structures from `get_stucture`, save them and then use these to re-create a `Trajectory` on HDF loading.

Since we're not using `to_hdf/from_hdf` yet, I'm not sure if it's worth the extra work.  In this case there's yet another way to consider: don't derive from `DataContainer` and make the caller of `NeighborsTraj` responsible for saving the output.  E.g. `AtomisticJob.get_neighbors` would create a `NeighborsTraj`, save the output to its own `self['output/neighbors/...'] then return the results.